ksql-engine/src/main/java/io/confluent/ksql/util/AvroUtil.java

```diff
@@ -41,7 +42,27 @@ public static boolean isValidSchemaEvolution(
 
     final String topicName = queryMetadata.getResultTopic().getKafkaTopicName();
 
-    return isValidAvroSchemaForTopic(topicName, avroSchema, schemaRegistryClient);
+    if (!isValidAvroSchemaForTopic(topicName, avroSchema, schemaRegistryClient)) {
+      final String subject = topicName + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX;
```

Comment:
Would you mind moving lines 45-54 into a method, e.g. `String getLatestSchema(topicName)`